src/main/java/io/cdap/plugin/gcp/common/GCPUtils.java

```diff
@@ -310,4 +314,54 @@ public static String formatAsFQNComponent(String component) {
       return component;
     }
   }
+
+  /**
+   * Get a ProgramFailureException with the given error information.
+   *
+   * @param errorReason The reason for the error.
+   * @param errorMessage The error message.
+   * @param errorType The type of error.
+   * @return A ProgramFailureException with the given error information.
+   */
+  public static ProgramFailureException getProgramFailureException(Exception cause,
```

Comment:
Mark cause as nullable. Also it's usual to have nullable items last, so that overloading can be introduced as needed

Reply:
done.